src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -15,4 +19,7 @@
     User save(User user);
 
     void deleteById(Long id);
+
+    @Query("SELECT u FROM User u WHERE lower(u.name) LIKE %:search% AND u.birthdate BETWEEN :from AND :to")
+    List<User> findBetweenDates(@Param("from")LocalDate from, @Param("to") LocalDate to, @Param("search") String search);
```

Comment:
Here it's not necessary to use a custom query.
You can take advantage of the findByBirthdateBetweenAndNameContainingAllIgnoreCase template ;)